### PR TITLE
fix: handle source node

### PR DESCRIPTION
**Pull request by AdiGajbhiye**
## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/manifest/parsers/graphParser.ts [hunk 1]
```diff
@@ -89,11 +89,7 @@ export class GraphParser {
           const url = sourceMetaMap
             .get(sourceName)
             ?.tables.find((table) => table.name === tableName)?.path!;
-          return new Source(
-            `${tableName} (${sourceName})`,
-            parentNodeName,
-            url,
-          );
+          return new Source(sourceName, parentNodeName, url);
```
Comment (mdesmet) [inline src/manifest/parsers/graphParser.ts:92]: Why this change?

File: src/webview_provider/newLineagePanel.ts [hunk 2]
```diff
@@ -252,19 +300,47 @@ export class NewLineagePanel implements LineagePanelView {
...
     }
+    const project = this.getProject();
+    if (!project) {
+      return;
+    }
     if (nodeType === DBTProject.RESOURCE_TYPE_SOURCE) {
       const { sourceMetaMap } = event;
       const splits = tableKey.split(".");
-      const schema = splits[2];
+      const sourceName = splits[2];
```
Comment (suryaiyer95) [inline src/webview_provider/newLineagePanel.ts:310]: Is there a case where length might be > 4?

**Comment by mdesmet** [comment]
Lineage is not working with sources

image

**Review by mdesmet — APPROVED** [review]
